src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -192,8 +192,8 @@ namespace Core::IO
       void operator()(ryml::NodeRef node, size_t* size)
       {
         FOUR_C_ASSERT(node.is_map(), "Expected a map node.");
-        // Pull up the "noneable" aspect. The fact that this type wraps another type is specific
-        // to C++ and not relevant to other tools. Simply knowing that a type can be "none" is
+        // Pull up the std::optional aspect. The fact that this type wraps another type is specific
+        // to C++ and not relevant to other tools. Simply knowing that a type can be empty is
         // enough for them.
         emit_value_as_yaml(node["noneable"], true);
```

Comment:
I talked to @gilrrei and we thought we just stick with the name `noneable` in the `metadata` for now. Writing `optional` here would be confusing since we also have a `required` field (only in the metadata!). Maybe we will come up with something better over time. In the end, it does not matter too much and the language barriers between C++, yaml, json schema, etc. will always require implicit knowledge.